packages/angular/src/library/jsonforms.component.ts

```diff
@@ -126,11 +127,32 @@ export class JsonFormsOutlet
       renderer !== undefined &&
       renderer.tester(uischema, schema, testerContext) !== -1
     ) {
-      bestComponent = renderer.renderer;
+      if (renderer.renderer instanceof Promise) {
+        renderer.renderer.then((resolvedRenderer) => {
```

Comment:
There can be a race condition here:
1. The outlet is invoked, however the renderer (Renderer A) is a promise and needs some time to complete
2. There is an update in the mean time, another renderer (Renderer B) is required and it is awaited too
3. The Renderer B resolves first and renders
4. Now finally Renderer A resolves and will be used for rendering, although Renderer B is the one which should be used.